gulpfile.js

```diff
@@ -1,14 +1,34 @@
 var copy = require("copy"),
     clear = require("cli-clear"),
     del = require("del"),
+    exec = require("child_process").exec,
     gulp = require("gulp"),
-    gutil = require("gulp-util");
+    gutil = require("gulp-util"),
+    runSequence = require("run-sequence");
 
 gulp.task("clear", function(cb) {
     clear();
     cb();
 });
 
+gulp.task("clean", function(cb) {
+    del("./coverage")
+    .then(function(files) {
+        if(files.length > 0) {
+            gutil.log("Deleted %s", files);
+        }
+        cb();
+    });
+});
+
+gulp.task("build", function (cb) {
+    exec("../../node_modules/.bin/tsc --project ../../tsconfig.json --rootDir ../../src", function (error, stdout, stderr) {
```

Comment:
Rather than manually invoking the `tsc` binary, I'd recommend using a Node.JS interface to typescript.
Looks like there's a gulp-specific solution with incremental compilation support :D https://github.com/ivogabe/gulp-typescript

Reply:
This does the job but it feels kind of hacky, I agree. My first approach was using the `npm run build` script but that script lives in the package.json of the root project and the gulp script executes as if it was inside each example/integration project (../../node_modules, ../../tsconfig.json etc) so I gave up that idea fairly quickly. I'll take a look at the gulp-typescript package ☺️

Reply:
Using gulp-typescript now 🚀